sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala

```diff
@@ -235,12 +236,13 @@ case class ExpressionEncoder[T](
 
     val plan = Project(Alias(unbound, "")() :: Nil, LocalRelation(schema))
     val analyzedPlan = SimpleAnalyzer.execute(plan)
+    val optimizedPlan = SimplifyCasts(analyzedPlan)
```

Comment:
I'm also thinking about if we should introduce `sqlContext` here, and use its analyzer and optimizer. For now our encoder resolution is case sensitive regardless of the `CASE_SENSITIVE` config.